content/automate/ManageIQ/Infrastructure/VM/Transform/Import.class/__methods__/import_infra_vm.rb

```diff
@@ -0,0 +1,49 @@
+#
+# Description: This method initiates VM import to given infra provider
+#
+
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Transform
+          module Import
+            class ImportInfraVm
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                vm = @handle.root['vm']
+                name = @handle.root['dialog_name']
+                provider_id = @handle.root['dialog_provider']
+                cluster_id = @handle.root['dialog_cluster']
+                storage_id = @handle.root['dialog_storage']
+                sparse = @handle.root['dialog_sparse']
+
+                provider = @handle.vmdb(:ext_management_system, provider_id)
+
+                provider.import_vm(
+                  vm.id,
+                  :name       => name,
```

Comment:
@matobet 
Could we not use the local variables and directly use the attribute values from root
```
provider.import_vm(
@handle.root['vm'].id,
:name =>  @handle.root['dialog_name'],
:cluster_id => @handle.root['dialog_cluster'],
:storage_id => @handle.root['dialog_storage'],
:sparse => @handle.root['dialog_sparse'])
```